File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int su=0;
        int smax = INT_MIN;
        for(int i: nums)
        {
            su+=i;
            smax = max(su,smax);
            if (su<0)   su =0;
        }
        return smax;
    }
};

File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public static int max(int a,int b)
        {
            if(a>b)
                return a;
            return b;
            
        }
    public int maxSubArray(int[] nums) {
        int sum =0;
        int smax = Integer.MIN_VALUE;
        
        for(int i: nums)
        {
            sum+=i;
            smax = max(sum,smax);
            sum = sum>0?sum:0;
        }
        return smax;
    }
}